src/installer.ts

```
@@ -323,9 +327,9 @@ async function ghcupBin(os: OS): Promise<string> {
   if (cachedBin) return join(cachedBin, 'ghcup');
 
   const bin = await tc.downloadTool(
-    `https://downloads.haskell.org/ghcup/${ghcup_version}/x86_64-${
-      os === 'darwin' ? 'apple-darwin' : 'linux'
-    }-ghcup-${ghcup_version}`
+    `https://downloads.haskell.org/ghcup/${ghcup_version}/${
+      arch === 'arm64' ? 'aarch64' : 'x86_64'
+    }-${os === 'darwin' ? 'apple-darwin' : 'linux'}-ghcup-${ghcup_version}`
```

Could that translation from option strings to arch names be extracted to its own function so that the translation is defined in exactly one place in the code?
(See comment above on code duplication.)

Also, this function should error out on an unknown `arch` rather than falling back to a default.  This would help users find errors.  E.g. if they misspelled `arm64` as `arm46` they would be alerted of the problem clearly.  
Update: maybe the type does not allow a value such as `arm46`,  but new architectures could be added by someone later, and not having a fallback would make clear that the translation has to be extended to the new value.